# HG changeset patch
# User Stefan Mahr <dac922@gmx.de>
# Date 1370367613 -7200
#      Tue Jun 04 19:40:13 2013 +0200
# Node ID c6d61dca5acd1e3edf6e4d51e37410ccc8b82bcf
# Parent  1c8b6ab2c8ae71f1fdef51016472396a6665f483
* strjoin.m: improve speed of joining long strings

diff --git a/scripts/strings/strjoin.m b/scripts/strings/strjoin.m
--- a/scripts/strings/strjoin.m
+++ b/scripts/strings/strjoin.m
@@ -67,17 +67,17 @@ function rval = strjoin (cstr, delimiter
     delimiter(end) = {""};
   elseif (numel (delimiter) != num - 1)
     error ("strjoin:cellstring_delimiter_mismatch",
       "strjoin: the number of delimiters does not match the number of strings")
   else
     delimiter(end+1) = {""};
   endif
 
-  rval = sprintf ("%s", [cstr(:).'; delimiter(:).']{:});
+  rval = [[cstr(:).'; delimiter(:).']{:}];
 
 endfunction
 
 %!assert (strjoin ({"hello"}, "-"), "hello")
 %!assert (strjoin ({"hello", "world"}), "hello world")
 %!assert (strjoin ({"Octave", "Scilab", "Lush", "Yorick"}, "*"),
 %!  "Octave*Scilab*Lush*Yorick")
 %!assert (strjoin ({"space", "comma", "dash", "semicolon", "done"},
